lib/Workflow.pm

```diff
@@ -291,8 +292,15 @@ sub _execute_single_action {
     my ( $new_state, $action_return );
 
     try {
-        $action->validate($self);
+        $action->validate($self, $action_args);
```

Comment:
@oliwel This is as you explained you're currently working with OpenXPKI. I'm wondering though: doesn't this cause tight integration between the call-site of the execute() method and the workflow implementation because different actions require different inputs - which they want to validate before executing the action. Having the action trigger validation on its arguments instead of on the workflow state means that the caller needs to pass exactly those arguments and there cannot be "preloading" of workflow state which the action will be re-using.

What's your opinion on this?

Reply:
This is very similar to the approach we use in OpenXPKI and I think this is a good approach to move forward with this.

Reply:
On Slack, we agreed that the actual implementation in OpenXPKI actually exactly addresses the issue I've tried to raise. I've changed the implementation accordingly.